### PR TITLE
Utility function for removing base filters

### PR DESCRIPTION
**Pull request by whikloj**
**JIRA Ticket**: https://jira.duraspace.org/browse/ISLANDORA-1838
# What does this Pull Request do?

If you set a base filter for Solr through the UI. You can then use the IslandoraSolrQueryProcessor to build your queries with appropriate XACML restrictions included but **NOT** those base filters.

For example:
We have a base filter of `-RELS_EXT_hasModel_uri_ms:"info:fedora/islandora:newspaperIssueCModel"` so newspaper issues (which aggregate the page OCR) don't display in search results.

But I wanted to use the `IslandoraSolrQueryProcessor` to generate a list of issues via Solr. To do this I had to remove the above filter from the parameters. 

I could keep this function in a separate module, but I have multiple uses for it and those places already depend on Islandora Solr. So it seemed the best place to avoid duplicating code.
# What's new?

Nothing, this function is not referenced in internal code (which might be a reason not to include it). But could be used by developers that might need this type of functionality.
# How should this be tested?

n/a
# Additional Notes:

Example:
- Does this change require documentation to be updated? no
- Does this change add any new dependencies? no
- Does this change require any other modifications to be made to the repository (ie. Regeneration activity, etc.)? no
- Could this change impact execution of existing code? no
# Interested parties

@jordandukart -  I'm not sure if it is acceptable to add this type of code to the core. So let me know what you think.

### CODE REVIEW

**Comment by whikloj** [comment]
@Islandora/7-x-1-x-committers bump

**Comment by jordandukart** [comment]
This slipped through the first time, looking at it now @whikloj

File: includes/utilities.inc
Comment (jordandukart) [inline includes/utilities.inc]: PHP should complain if it gets passed a string think this `is_array` check is not needed.

**Review by jordandukart — CHANGES_REQUESTED** [review]
Just the one thing, other than that looks good to me!

**Comment by jordandukart** [comment]
Will merge when this passes Travis!

**Review by jordandukart — APPROVED** [review]

**Pull request merged**
